src/components/app/App.tsx

```diff
@@ -0,0 +1,128 @@
+import { ReactElement, useState, useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { PAGE_TITLE, getAuthors, PAGE_NUMBER_DEFAULT } from '../../utils/data';
+import { httpGet } from '../../utils/httpGet';
+import Sidebar from '../sidebar/Sidebar';
+import Results from '../results/Results';
+import Spinner from '../spinner/Spinner';
+import type { SectionsResponseResults, SearchResponse, Author } from '../../types/types';
+
+export default function App(): ReactElement {
+    const [sections, setSections] = useState<SectionsResponseResults[] | null>(null);
+    const [articles, setArticles] = useState<SearchResponse | null>(null);
+    const [authors, setAuthors] = useState<Author[]>(getAuthors());
+    const [searchParams, setSearchParams] = useSearchParams();
+    const [isLoadingArticles, setIsLoadingArticles] = useState<true | false>(false);
+    const [isLoadingSections, setIsLoadingSections] = useState<true | false>(false);
+    const [hasErrorArticles, setHasErrorArticles] = useState<true | false>(false);
+    const [hasErrorSections, setHasErrorSections] = useState<true | false>(false);
+    // const [state, setState] = useState<any[]>([]);
+
+    // useEffect(() => {
+    //     setState([...state, searchParams]);
+    // }, [searchParams])
+
+    // useEffect(() => {
+    //     state.forEach((value, idx, array) => {
+    //         if (array.length - 1 === idx) return;
+    //         console.log(value === array[idx + 1]);
+    //         console.log(state);
+    //     })
+    // }, [state]);
+
+
+    function handleSubmit(searchPhrase: string): void {
+        if (!searchPhrase) return;
+        searchParams.set('q', searchPhrase);
+        searchParams.set('page', PAGE_NUMBER_DEFAULT);
+        setSearchParams(searchParams);
+    }
+
+    function handleReset() {
+        //__to show /?page=1
+        setSearchParams({ page: PAGE_NUMBER_DEFAULT });
```

Comment:
I think it results in wrong UX -> I would maybe go with `All categories` instead `All articles` and wouldn't change `q` param in that case.
And for reseting search totally maybe even separate button would be better - up to you, not critical.